elide-core/src/main/java/com/yahoo/elide/Elide.java

```diff
@@ -77,6 +92,69 @@ public Elide(ElideSettings elideSettings) {
         elideSettings.getSerdes().forEach((targetType, serde) -> {
             CoerceUtil.register(targetType, serde);
         });
+
+        registerCustomSerde();
+        registerCustomSerdeInObjectMapper();
+    }
+
+    private void registerCustomSerde() {
```

Comment:
I refactored this a bit by:
  1. Using a different member of ClassScanner (see other comment in this PR).
  2. Linking the methods together to reduce some duplication between them.

```java
    private void registerCustomSerde() {
        Set<Class<?>> classes = ClassScanner.getAnnotatedClasses(ElideTypeConverter.class);

        for (Class<?> clazz : classes) {
            if (!Serde.class.isAssignableFrom(clazz)) {
                log.warn("Skipping Serde registration (not a Serde!): {}", clazz);
                continue;
            }
            Serde serde;
            try {
                serde = (Serde) clazz.newInstance();
            } catch (InstantiationException | IllegalAccessException e) {
                String errorMsg = String.format("Error while registering custom Serde: %s", e.getLocalizedMessage());
                log.error(errorMsg);
                throw new UnableToAddSerdeException(errorMsg);
            }
            ElideTypeConverter converter = clazz.getAnnotation(ElideTypeConverter.class);
            Class baseType = converter.type();
            List<Class<?>> allTypes = Arrays.asList(converter.subTypes());
            allTypes.add(baseType);

            for (Class type : allTypes) {
                if (!baseType.isAssignableFrom(type)) {
                    throw new IllegalArgumentException("Mention type " + type
                            + " not subtype of " + baseType);
                }
                log.info("Registering serde for type : {}", type);
                CoerceUtil.register(type, serde);
                registerCustomSerdeInObjectMapper(type, serde, converter.name());
            }
        }
    }
    private void registerCustomSerdeInObjectMapper(Class<?> type, Serde serde, String name) {
        ObjectMapper objectMapper = mapper.getObjectMapper();
        objectMapper.registerModule(new SimpleModule(name)
            .addSerializer(type, new JsonSerializer<Object>() {
                @Override
                public void serialize(Object obj, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
                        throws IOException, JsonProcessingException {
                    jsonGenerator.writeObject(serde.serialize(obj));
                }
            }));
    }
```

Reply:
Replaced my code with yours, made some minor changes here and there. Thanks! for the lesson on short circuit though :)